relay-server/src/processing/common.rs

```diff
@@ -60,3 +71,127 @@ impl From<Nothing> for Outputs {
         match value {}
     }
 }
+
+/// Shared metadata extraction utilities for trace items.
+///
+/// This module contains common functionality for extracting metadata from
+/// annotated structures and converting them to TraceItem attributes.
+#[cfg(feature = "processing")]
+pub mod meta_extraction {
+    use super::*;
+
+    /// Represents metadata extracted from Relay's annotated model.
+    ///
+    /// This struct holds metadata about processing errors, transformations, and other
+    /// information that occurred during processing of the original payload.
+    ///
+    /// The attribute metadata itself is serialized as a JSON string.
+    #[derive(Debug, Serialize)]
+    pub struct AttributeMeta {
+        /// Meta as it was extracted from Relay's annotated model.
+        pub meta: MetaTree,
+    }
+
+    impl AttributeMeta {
+        /// Converts the metadata to an AnyValue for TraceItem attributes.
+        ///
+        /// Returns None if the metadata is empty, otherwise serializes the metadata
+        /// to JSON and wraps it in a StringValue.
+        pub fn to_any_value(&self) -> Option<AnyValue> {
+            if self.meta.is_empty() {
+                return None;
+            }
+
+            let s = serde_json::to_string(self)
+                .inspect_err(|err| {
+                    relay_log::error!(
+                        error = err as &dyn std::error::Error,
+                        "attribute meta serialization failed"
+                    )
+                })
+                .ok()?;
+
+            Some(AnyValue {
+                value: Some(any_value::Value::StringValue(s)),
+            })
+        }
+    }
+
+    /// Extracts TraceItem meta attributes from any structure that implements IntoValue.
+    ///
+    /// The implementation piggy backs on [`IntoValue::extract_child_meta`],
+    /// a lighter implementation using a [`relay_event_schema::processor::Processor`]
+    /// which removes the meta instead of cloning.
```

Comment:
There was some texdt in this comment I think was accidental, removed 

> /// which removes the meta instead of cloning it is thinkable.


https://github.com/getsentry/relay/blob/master/relay-server/src/processing/logs/store.rs#L137